### ISSUE TITLE
Unable to make samplesubscribe.js (Node.JS patterns) to work as expected

### ISSUE THREAD

**Issue by creasman** [ibm-messaging/mq-dev-patterns#13]
My goal is to use the mq-mqi-nodejs (ibmmq) package to subscribe to an MQ topic with a callback that is invoked whenever a message arrives to the topic.  From reading the Node.js documentation in this package my understanding is that is what the samplesubscribe.js example is doing.  

I followed the coding example as shown in samplesubscribe.js and boilerplate.js to implement this function in my MQ application.  What I noticed was that the first message sent would be received and processed, but no more after that.  I used IBM MQ Explorer to verify the messages were being sent to the topic by setting up a test subscription there.  The test subscription received all messages sent to the topic.

Thinking I had implemented the code incorrectly, I ran the samplesubscribe.js sample directly -- only pointing to my queue manager and topic.  Again, only the first message was received and after that none.  

I enabled the debug tracing in the mq-mqi-nodejs client to see what it was doing.  In there I see the polling loop continues to run as expected.  However, it only shows the first message being received.  My hunch is the problem may lie inside the ibmmq client code and not in these patterns.  I'm opening the issue here as a start, though, in case I am misunderstanding the intent of this sample.

Again, my goal is to code my application using the ibmmq client such that I can provide a callback at the time I subscribe.  The callback would then be called whenever a new message(s) arrive.  I can implement a polling loop inside the application but understood this is something the ibmmq client will do if coded correctly.  

Has anyone else got this sample to work as stated?  Am I misunderstanding the intent here?  Please advise.

**Comment by chughts** [comment]
I will investigate.

**Comment by chughts** [comment]
This is because there is a bug in `boilerplate.js` when it is setting up the MQGMO object. 
The line 

```node.js
      if ('GET' === me.modeType) {
        gmo.MatchOptions = MQC.MQMO_NONE;
      } 
```

should be 

```node.js
      if ('GET' === me.modeType || 'SUBSCRIBE' === me.modeType) {
        gmo.MatchOptions = MQC.MQMO_NONE;
      } 
```

I will create a pull request with the fix.

**Comment by chughts** [comment]
Fixed in #14

**Comment by creasman** [comment]
Thanks!  It works after changing the default option value.
